eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java

```diff
@@ -140,7 +153,57 @@ public void failsOnDuplicateNonIdenticalVariant() throws Exception {
         DbsnpClusteredVariantEntity duplicateVariant = new DbsnpClusteredVariantEntity(
                 EXPECTED_ACCESSION, hashingFunction.apply(duplicateClusteredVariant), duplicateClusteredVariant);
 
-        thrown.expect(RuntimeException.class);
-        dbsnpClusteredVariantWriter.write(Arrays.asList(variant, duplicateVariant));
+        thrown.expect(BulkOperationException.class);
+        try {
+            dbsnpClusteredVariantWriter.write(Arrays.asList(variant, duplicateVariant));
+        } finally {
+            assertEquals(1, importCounts.getClusteredVariantsWritten());
+        }
+    }
+
+    @Test
+    public void allNonDuplicatedRecordsWillBeWritten() {
+        // batch of 6 variants, 5 unique and 1 duplicate that is in the middle of the batch
+        ClusteredVariant firstClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                 VARIANT_TYPE, false);
+        ClusteredVariant secondClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_2, "contig", START_1,
+                                                                       VARIANT_TYPE, VALIDATED);
+        ClusteredVariant thirdClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig_3", START_1,
+                                                                       VARIANT_TYPE, VALIDATED);
+        ClusteredVariant duplicateFirstClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
```

Comment:
You could use `firstClusteredVariant` to construct 2 entities, which will make sure that there are no bugs due to typos etc.